### PR TITLE
Kemboi

### PR DESCRIPTION
**Pull request by kemboi22**
- Few changes in dashboard cards
Screenshot From 2024-12-04 01-50-24

- Payment Receipt

> Added emailing of payment receipt after payment 

# TODO
1.  In the payment receipt incude donloading PDF and adding receipt number(Mpesa Transaction Code)

### CODE REVIEW

File: app/pages/dashboard.vue
Comment (kgarchie) [inline app/pages/dashboard.vue:1]: Good work!

Though next time, you could use a pre-commit formatting hook/strategy so that only the code you have diffed is formatted. This reduces the number of unnecessary changes to be looked at.

File: app/pages/dashboard.vue [hunk 1]
```diff
@@ -1,246 +1,276 @@
...
-        <ul class="w-full xl:w-6/12 xl:px-5 lg:px-10 lg:w-full md:w-9/12 m-auto mt-2 text-center">
-            <li class="py-2 bg-emerald-700 text-white rounded hover:bg-emerald-600 transition-colors w-40 font-semibold uppercase"
-                style="font-size: smaller;">
-                <NuxtLink to="/forms/create">Create A Form</NuxtLink>
-            </li>
-        </ul>
-    </div>
-</template>
 <script setup lang="ts">
+import { DollarSignIcon, FileTextIcon, GitGraphIcon } from 'lucide-vue-next';
```
Comment (kgarchie) [inline app/pages/dashboard.vue:2]: I see the library is tree-shakeable. I assume it won't add too much weight to the bundle. That's good.

File: app/pages/dashboard.vue [hunk 2]
```diff
@@ -1,246 +1,276 @@
...
+  {
+    title: "REACH",
+    count: stats?.responses || 0,
+    url: "/finance",
+    description: `<span>You have reached</span>
+<span class="font-medium text-gray-900 mx-1">${stats?.responses || 0}</span>
+              <span>people so far</span>`,
+    icon: GitGraphIcon
+  },
+];
```
Comment (kgarchie) [inline app/pages/dashboard.vue:68]: I avoid writing HTML in TS and leave it in the Templates. Transform this into a component instead.

```
defineComponent({...})
```

Check out the docs [here](https://vuejs.org/api/general.html#definecomponent).

**Review by kgarchie — CHANGES_REQUESTED** [review]
Just some few things to note before merging. Solid.

File: app/pages/dashboard.vue
Comment (kemboi22) [reply app/pages/dashboard.vue:1]: Its my editor formatting the docs

File: app/pages/dashboard.vue (hunk 1 above)
Comment (kemboi22) [reply app/pages/dashboard.vue:2]: Yeah It is

File: app/pages/dashboard.vue (hunk 2 above)
Comment (kemboi22) [reply app/pages/dashboard.vue:68]: I Dont think i can define a component within the script setup

File: app/pages/dashboard.vue
Comment (kemboi22) [reply app/pages/dashboard.vue:1]: Okay let me see it

**Review by kemboi22 — COMMENTED** [review]
I think you should define the prettier config and eslint json so as to have the same formatting

File: app/pages/dashboard.vue [hunk 3]
```diff
@@ -1,246 +1,214 @@
...
+          h('h3', { class: "text-2xl font-bold text-gray-900 mt-1" }, this.$props?.count || 0)
+        ]),
+        h('div', { class: "bg-gray-100 group-hover:bg-gray-200 rounded-full p-3 transition-colors" }, [
+          h(this.$props.icon, { class: "w-6 h-6 text-gray-700" })
+        ])
+      ]),
+      h('div', { class: "mt-4 flex items-center text-sm text-gray-600" }, this.$props.description)
+    ])
+  }
+})
```
Comment (kgarchie) [inline app/pages/dashboard.vue:55]: This is how to define a component within a component programmatically without having to create a new file

File: package.json
Comment (kgarchie) [inline package.json:1]: Before
image

After
image

Unless this library is utilized elsewhere, it will be removed in the future. It's too expensive for only 3 usage instances..

**Review by kgarchie — APPROVED** [review]
